.github/actions/maven-build-and-tag/action.yml

```diff
@@ -229,3 +229,17 @@ runs:
       run: |
         git tag -a $VERSION -m "Release version $VERSION"
         git push origin $VERSION
+
+    - name: Clean m2 cache
+      if: github.event_name == 'push'
+      shell: bash
+      run: |
+        rm -fr ~/.m2/repository/org/activiti
+        rm -fr ~/.m2/repository/com/alfresco
+
+    - name: Save maven cache
+      uses: actions/cache/save@704facf57e6136b1bc63b828d79edcd491f0ee84 # v3.3.2
+      if: github.event_name == 'push'
```

Comment:
this doesn't prevent saving cache for branches different than the default one if the caller workflow is configured to build also those

something like `&& github.event.repository.default_branch == github.ref_name` should fix that

Reply:
thanks!

Reply:
Actually I think we want to keep the cache on maintenance branches, that's why keeping the logic checking event `push` makes more sense to me

Reply:
Yes, I think we need to consider maintenance branches as well, and we usually don't configure the CI to build on feature branches, so it should not be a problem.